tests/sklearn/test_sklearn_autolog.py

```diff
@@ -668,56 +668,55 @@ def test_autolog_emits_warning_message_when_model_prediction_fails():
         # Count how many times `mock_warning` has been called on not-fitted `predict` failure
         call_count = len([args for args in mock_warning.call_args_list if msg in args[0][0]])
         # If `_is_plotting_supported` returns True (meaning sklearn version is >= 0.22.0),
-        # `mock_warning` should have been called twice, once for metrics, once for artifacts.
-        # Otherwise, only once for metrics.
-        call_count_expected = 2 if mlflow.sklearn.utils._is_plotting_supported() else 1
+        # `mock_warning` should have been called three times: once for generic estimator metrics,
+        # once for the `training_score` metric, and once for artifacts. Otherwise, just twice for
+        # generic estimator metrics and the `training_score` metric
+        call_count_expected = 3 if mlflow.sklearn.utils._is_plotting_supported() else 2
         assert call_count == call_count_expected
 
 
 def test_fit_xxx_performs_logging_only_once(fit_func_name):
-    mlflow.sklearn.autolog()
+    with mock.patch(
+        "mlflow.sklearn.utils._log_pretraining_metadata"
+    ) as mock_log_pretraining, mock.patch(
+        "mlflow.sklearn.utils._log_posttraining_metadata"
+    ) as mock_log_posttraining:
+        mlflow.sklearn.autolog()
 
-    model = sklearn.cluster.KMeans()
-    X, y = get_iris()
+        model = sklearn.cluster.KMeans()
+        X, y = get_iris()
 
-    with mock.patch("mlflow.log_params") as mock_log_params, mock.patch(
-        "mlflow.log_metric"
-    ) as mock_log_metric, mock.patch("mlflow.set_tags") as mock_set_tags, mock.patch(
-        "mlflow.sklearn.log_model"
-    ) as mock_log_model:
```

Comment:
This kind of test becomes brittle / hard to interpret as we change which APIs are used to log content in an effort to batch things more effectively. Instead, we refactor this test case to check that `mlflow.sklearn.utils._log_pretraining_metadata` and `mlflow.sklearn.utils._log_posttraining_metadata` are called once. See https://github.com/mlflow/mlflow/pull/4259/files#r616143381 for more context.